// @soroban-react/chains/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.standalone = exports.sandbox = exports.testnet = exports.futurenet = exports.mainnet = void 0;
const stellar_sdk_1 = require("@stellar/stellar-sdk");
/**
 * A `WalletChain` object representing the public blockchain network.
 * @property {string} id - The unique identifier for the blockchain network.
 * @property {string} name - The name of the blockchain network.
 * @property {string} network - The name of the blockchain network.
 * @property {string} networkPassphrase - The network passphrase for the blockchain network.
 * @property {string} networkUrl - The Horizon URL "https://horizon-testnet.stellar.org";
 * @property {string} sorobanRpcUrl  - The Soroban RPC URL sorobanRpcUr;
 */
exports.mainnet = {
    id: 'mainnet',
    network: 'mainnet',
    name: 'Mainnet',
    networkPassphrase: stellar_sdk_1.Networks.PUBLIC,
    networkUrl: 'https://horizon.stellar.org',
    sorobanRpcUrl: 'https://mainnet.stellar.validationcloud.io/v1/XFb5Lma6smizBnnRPEgYMbuNm0K3FWzJ7854GNSQ2EY',
};
/**
 * A `WalletChain` object representing the Futurenet blockchain network.
 */
exports.futurenet = {
    id: 'futurenet',
    network: 'futurenet',
    name: 'Futurenet',
    networkPassphrase: 'Test SDF Future Network ; October 2022',
    networkUrl: 'https://horizon-futurenet.stellar.org',
    sorobanRpcUrl: 'https://rpc-futurenet.stellar.org/',
};
/**
 * A `WalletChain` object representing the Testnet blockchain network.
 */
exports.testnet = {
    id: 'testnet',
    network: 'testnet',
    name: 'Testnet',
    networkPassphrase: stellar_sdk_1.Networks.TESTNET,
    networkUrl: 'https://horizon-testnet.stellar.org',
    sorobanRpcUrl: 'https://soroban-testnet.stellar.org/',
};
/**
 * A `WalletChain` object representing the Sandbox blockchain network.
 */
exports.sandbox = {
    id: 'sandbox',
    network: 'sandbox',
    name: 'Sandbox',
    networkPassphrase: 'Local Sandbox Stellar Network ; September 2022',
    networkUrl: 'http://localhost:8000',
    sorobanRpcUrl: 'http://localhost:8000/soroban/rpc',
};
/**
 * A `WalletChain` object representing the Standalone blockchain network.
 */
exports.standalone = {
    id: 'standalone',
    network: 'standalone',
    name: 'Standalone',
    networkPassphrase: 'Standalone Network ; February 2017',
    networkUrl: 'http://localhost:8000',
    sorobanRpcUrl: 'http://localhost:8000/soroban/rpc',
};
